docs/001_develop/03_client-capabilities/008_interaction/checkbox.mdx

```diff
@@ -0,0 +1,180 @@
+---
+title: 'Checkbox'
+sidebar_label: 'Checkbox'
+id: client-interaction-checkbox
+keywords: [interaction, interactive, checkbox, box, boolean, form, click, change, checked, toggle, submit, data]
+tags:
+- interaction
+- interactive
+- checkbox
+- box
+- boolean
+- form
+- click
+- change
+- checked
+- toggle
+- submit
+- data
+sidebar_position: 8
+---
+
+import CheckboxDemo from '../../../../examples/ui/client-capabilities/interaction/checkbox.js';
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+# Checkbox
+
+A form-connected implementation of a [checkbox](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/checkbox) input.
+
+Use cases:
+* Multi-selection forms
+* Filtering
+* Sorting
+* To-Do lists
+* Privacy and consent
+
+Any use where the user is to give a boolean type input choice.
+
+## Example
+
+<CheckboxDemo />
+
+<Tabs defaultValue="genesis" values={[{ label: 'Genesis', value: 'genesis', }, { label: 'React', value: 'react', }, { label: 'Angular', value: 'angular', }]}>
+
+<TabItem value="genesis">
+
+Declaration
+```typescript
+<rapid-checkbox></rapid-checkbox>
+```
+Usage
+```typescript
+@customElement({
+  name: 'my-element',
+  template: html`
+        <rapid-checkbox
+            ?disabled="${(x) => x.disabled}"
+            ?checked="${sync((x) => x.isSelected, 'boolean')}"
+            @change="${(x) => x.checkChanged()}"
+        >Checkbox
+        </rapid-checkbox>
+    `,
+})
+export class MyElement extends GenesisElement {
+  @observable disabled = false;
+  @observable isSelected = false;
+  checkChanged() {
+    console.log(this.isSelected);
+  }
+}
+```
+
+</TabItem>
+<TabItem value="react">
+
+Declaration
+```typescript
+<rapid-checkbox></rapid-checkbox>
+```
+Usage
+```typescript
+export function MyComponent() {
+	const [checkboxValue, setCheckboxState] = useState(false);
+	const handleCheckboxChange = (event) => {
+		setCheckboxState(!!event.target.checked);
+		console.log({ checkboxValue })
+	};
+
+	return (
+		<rapid-checkbox onChange={handleCheckboxChange}></rapid-checkbox>
+	);
+}
+```
+</TabItem>
+<TabItem value="angular">
+
+Declaration
+```typescript
+<rapid-checkbox></rapid-checkbox>
+```
+
+Model
+```typescript my.component.ts
+import { Component, CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
+import { FormsModule } from '@angular/forms';
+@Component({
+  selector: 'my-root',
+  templateUrl: './my.component.html',
```

Comment:
Here’s a way to use an inline template like:
```
template: `<rapid-checkbox
name="isSelected"
[checked]="isSelected"
(change)="onCheckboxChange($event)"
></rapid-checkbox>` 
```

If we used this in this way, it would be consistent across all frameworks (so the examples would be single-file).

Reply:
Thanks for the info, I have adjusted it to be the same!